src/components/SingleSelect/SingleSelect.tsx

```diff
@@ -22,96 +36,179 @@ const {
+	description?: string;
+}
+
+const Placeholder = (_props: ISingleSelectPlaceholderProps): null => null;
+
+Placeholder.displayName = 'SingleSelect.Placeholder';
+Placeholder.peek = {
+	description: `
+		Content this is displayed when nothing is selected.
+	`,
+};
+Placeholder.propName = 'Placeholder';
+
+/** Option Child Component */
+export interface ISingleSelectOptionProps extends IDropMenuOptionProps {
+	description?: string;
+	name?: string;
+	/** Custom Option component (alias for `SingleSelect.Option.Selected`)  */
+	Selected?: React.ReactNode;
+}
+
+const Selected = (_props: { children?: React.ReactNode }): null => null;
+
+Selected.displayName = 'SingleSelect.Option.Selected';
+Selected.peek = {
+	description: `
+		Customizes the rendering of the Option when it is selected
+		and is displayed instead of the Placeholder.
+	`,
+};
+Selected.propName = 'Selected';
+Selected.propTypes = {};
+
+const Option = (_props: ISingleSelectOptionProps): null => null;
+
+Option.displayName = 'SingleSelect.Option';
+Option.peek = {
+	description: `
+		A selectable option in the list.
+	`,
+};
+Option.Selected = Selected;
+Option.propName = 'Option';
+Option.propTypes = {
+	Selected: any`
+		Customizes the rendering of the Option when it is selected and is
+		displayed instead of the Placeholder.
+	`,
+	...DropMenu.Option.propTypes,
+};
+Option.defaultProps = DropMenu.Option.defaultProps;
+
+const OptionGroup = (_props: IDropMenuOptionGroupProps): null => null;
+
+OptionGroup.displayName = 'SingleSelect.OptionGroup';
+OptionGroup.peek = {
+	description: `
+		Groups \`Option\`s together with a non-selectable heading.
+	`,
+};
+OptionGroup.propName = 'OptionGroup';
+OptionGroup.propTypes = DropMenu.OptionGroup.propTypes;
+OptionGroup.defaultProps = DropMenu.OptionGroup.defaultProps;
+
+type ISingleSelectDropMenuProps = Partial<IDropMenuProps>;
+
+/** Single Select Component */
+interface ISingleSelectPropsRaw extends StandardProps {
+	/** Custom Placeholder component (alias for `SingleSelect.Placeholder`)  */
+	Placeholder?: React.ReactNode;
+
+	/** Custom Option component (alias for `SingleSelect.Option`)  */
+	Option?: React.ReactNode;
+
+	//TODO: Remove? Seems like this belongs on OptionProps not SingleSelectProps
+	/** Custom Option component (alias for `SingleSelect.Option.Selected`)  */
+	Selected?: React.ReactNode;
+
+	/** Custom OptionGroup component (alias for `SingleSelect.OptionGroup`)  */
+	OptionGroup?: IDropMenuOptionGroupProps;
+
+	hasReset: boolean;
+
+	isSelectionHighlighted: boolean;
+
+	isDisabled: boolean;
+
+	isInvisible: boolean;
+
+	selectedIndex: number | null;
+
+	//DropMenu: IDropMenuProps;
+	DropMenu: ISingleSelectDropMenuProps;
+
+	maxMenuHeight?: number | string;
+
+	onSelect?: (
+		optionIndex: number | null,
+		{
+			props,
+			event,
+		}: {
+			props: ISingleSelectOptionProps | undefined;
+			event: React.MouseEvent | React.KeyboardEvent;
+		}
+	) => void;
+
+	/** TODO: doublecheck this type */
+	ref?: string;
```

Comment:
This does indeed seem weird. Was it needed for some reason?

Reply:
This was a tricky one.

Since we were using `Overwrite` for the `ISingleSelectPropsRaw` interface, typescript seemed to think we wanted a `div` ref prop, not a `SingleSelect` ref, causing this line to give a type error: https://github.com/appnexus/lucid/blob/65d922390e79d8f1fc5320e368a4bd1b710ae472/src/components/Paginator/Paginator.jsx#L180

Looking at it again, I don't think we should've been using Overwrite for SingleSelect in the first place since `SingleSelect` doesn't pass through props to any native HTML tags. I removed it, and now everything typechecks without declaring `ref` in `ISingleSelectProps`.